front/lib/api/assistant/generation.ts

```diff
@@ -410,6 +433,32 @@ export async function* runGeneration(
       return;
     }
 
+    const currentTimestamp = Date.now();
+    if (currentTimestamp - lastCheckCancellation >= 1000) {
```

Comment:
Let's make the 1000 a visible `const CANCELLATION_CHECK_INTERVAL` and set it to 500 maybe?